src/types.rs

```diff
@@ -89,10 +114,21 @@ pub enum MessageKind {
         type_code: u8,
         kind: ADSBMessageKind,
     },
+    ModeSMessage {
+        icao_address: ICAOAddress,
+        kind: ModeSMessageKind,
+    },
     /// Unsupported message
     Unknown,
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub enum ModeSMessageKind {
+    // DF=5
+    SurveillanceIdentity { squawk: Squawk },
+    Unknown,
```

Comment:
Parsing of unsupported messages, both ADSB messages and these new Mode-S messages, is already caught by `MessageKind::Unknown` so this case isn't required.